ms-common-impl/src/main/java/net/trajano/ms/vertx/beans/BeanValidator.java

```diff
@@ -0,0 +1,173 @@
+import org.springframework.stereotype.Component;
+
+import net.trajano.ms.core.ConstraintViolationResponse;
+import net.trajano.ms.core.ErrorResponses;
+
+/**
+ * This performs validation checks on the request message. If the request
+ * message does not pass validation it will throw a bad request exception.
+ *
+ * @author Archimedes Trajano
+ */
+@Component
+public class BeanValidator implements
+    GeneralValidator {
+
+    /**
+     * Logger.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(BeanValidator.class);
+
+    @Autowired
+    private Validator validator;
+
+    @Override
+    public void checkViolations(final HttpRequest request) {
+
+        // does nothing
+
+    }
+
+    /**
+     * Obtain the name from JAX-RS annotations. If the parameter is not annotated is
+     * uses "body" as the name.
+     *
+     * @param parameter
+     *            parameter
+     * @return parameter name
+     */
+    private String getParameterName(final Parameter parameter) {
+
+        final String name;
+        if (parameter.getAnnotation(FormParam.class) != null) {
+            name = "form." + parameter.getAnnotation(FormParam.class).value();
+        } else if (parameter.getAnnotation(QueryParam.class) != null) {
+            name = "query." + parameter.getAnnotation(QueryParam.class).value();
+        } else if (parameter.getAnnotation(PathParam.class) != null) {
+            name = "path." + parameter.getAnnotation(PathParam.class).value();
+        } else if (parameter.getAnnotation(HeaderParam.class) != null) {
+            name = "header." + parameter.getAnnotation(HeaderParam.class).value();
+        } else if (parameter.isAnnotationPresent(Context.class)) {
+            name = parameter.getType().getName();
+        } else {
+            name = "body";
+        }
+        return name;
+    }
+
+    @Override
+    public boolean isMethodValidatable(final Method method) {
+
+        return true;
+    }
+
+    @Override
+    public boolean isValidatable(final Class<?> clazz) {
+
+        return true;
+    }
+
+    @Override
+    public void validate(final HttpRequest request,
+        final Object object,
+        final Class<?>... groups) {
+
+        LOG.debug("validate {} {}", request, object);
+
+    }
+
+    /**
+     * {@inheritDoc}. {@link Context} annotated parameters are skipped unless the
+     * {@link Valid} annotation is present.
+     */
+    @Override
+    public void validateAllParameters(final HttpRequest request,
+        final Object object,
+        final Method method,
+        final Object[] parameterValues,
+        final Class<?>... groups) {
+
+        LOG.debug("validateAllParameters {} {} {}", request, object, parameterValues);
+
+        final Map<String, Set<ConstraintViolation<Object>>> violationMap = new LinkedHashMap<>();
+        boolean hasViolation = false;
+        for (int i = 0; i < method.getParameterCount(); ++i) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS135)